### PR TITLE
Add bwtester dials and tabs GUI

### PR DESCRIPTION
**Pull request by mwfarb**
- Added tabs for each app.
- Moved client/server input into color static svg/inputs.
- Moved css and js into separate files.
- Added color arrows showing directionality of tests/content.
- Moved console output into scrollable window.
- Added bwtester tab with dials for each parameter/direction.
- Fixed bwtester command arg formatting issue.
- Added bwtester radio icon lock for each parameter.
- Added bwtester feedback to alter non-locked dials on change.
- Added bwtester feedback when change would exceed max on non-locked
dial.
- Added camerapp content over arrow, scaled to fit.
- Added camerapp ability to click and expand image.
- Added button disable when app is running.
- Updated readme with new directions and screenshots.
- Added tooltip note over args input about formatting.

### CODE REVIEW

**Comment by mwfarb** [comment]
@FR4NK-W @chaehni Let me know how this works for you.

**Comment by chaehni** [comment]
@mwfarb It works really well! Some feedback/ideas for improvement:

- the `packets` dial doesn't seem to work properly for me. When I lock the packet size and try to turn the packets dial it shows huge numbers and then springs back to its original value.
- When I click the image on the `camerapp` tab the image does not expand.
- When running the `bwtester`, in the console the app prints the server-client direction as extra string while the client-server direction is not printed as extra string. Is this intended?
- Can you add some form validation for the client/server addresses to the web interface?

**Comment by mwfarb** [comment]
@chaehni Sorry to add another commit to your review. The scionlab environment appears not to have `$GOBIN` set or run `go install` on setup, so the scionlab apps may not be compiled for new users. Also, I was mistakenly launching `go run` for each app run, causing extra compilation time on each run. The majority of this change is to ensure the binaries build on startup, and using the binary only on each app run.

**Comment by chaehni** [comment]
@mwfarb I had another look and my comments from above are addressed.
One idea: Can we limit the range of the dials based on the locked value and the limit of the remaining dial such that it cannot exceed any limits?
So if we, for example, lock the packet size the packets dial would be limited to a value that does not exceed the bandwidth limit of 150mbps. Otherwise, it's really hard to set a value for the packets via the dial.

I like that the apps are now installed instead of compiled for every run. I talked to Juan and we want to pre-install the apps in the VMs and run your webapp automatically. So we might only need to check if the apps are installed. (see https://github.com/netsec-ethz/scion-coord/issues/228)

**Comment by mwfarb** [comment]
The knob library has a non-standard way of displaying the text input box. Have you already tried to click on the number of packets and type in the number you want? If that is the root of setting the number of packets, I should change the display to look more like a standard input box.

**Comment by chaehni** [comment]
I didn't know you can enter it like this. 
That makes it a lot more convenient!

**Comment by mwfarb** [comment]
@chaehni I'd like to update the dial visually in another PR. Ultimately I'd like hash marks for the range of values on the dial, and an input box that is more intuitive to edit. For now, I've added a line of instruction above the dials making it clear that numbers can be typed, edited, clicked, or scrolled to change. Integrating a new dial library now may drag use of other improvements out more weeks.

**Comment by chaehni** [comment]
@mwfarb Thanks for your work. I think the instructions are good enough for now. We don't need to change the dials in this PR.

File: webapp/webapp.go [hunk 1]
```diff
@@ -15,51 +15,160 @@ import (
...
+    }
+    jsonBlob = append(jsonBlob, []byte(` ] }`)...)
+    err = ioutil.WriteFile(cli_fp, jsonBlob, 0644)
+    if err != nil {
+        log.Println("ioutil.WriteFile() error: " + err.Error())
+    }
+}
+
 // Handles loading index.html for user at root.
 func mainHandler(w http.ResponseWriter, r *http.Request) {
```
[the remark this answers is not in the document]
Comment (chaehni) [reply webapp/webapp.go]: The same happens in the image handler when the file is not found. I think we should respond with an appropriate error code.

**Comment by mwfarb** [comment]
OK, sounds good. I've removed the unused code and added the error messaging.

File: webapp/webapp.go [hunk 2]
```diff
@@ -172,6 +171,7 @@ func mainHandler(w http.ResponseWriter, r *http.Request) {
     data, err := ioutil.ReadFile(filepath)
     if err != nil {
         log.Fatal("ioutil.ReadFile() error: " + err.Error())
+        fmt.Fprint(w, "Error: Unable to read "+filepath)
```
[the remark this answers is not in the document]
Comment (mwfarb) [reply webapp/webapp.go]: Great! Thanks for the tip to use `http.Error()` as well. Let me know if there are any more issues before merging.

[the remark this answers is not in the document]
Comment (chaehni) [reply webapp/webapp.go]: Looks good to me.
I think we can merge it.

**Pull request merged**